src/test/kotlin/org/rust/lang/utils/RsEscapesUtilsTest.kt

```diff
@@ -57,3 +62,47 @@ class UnescapeRsTest(private val input: String,
         )
     }
 }
+
+@RunWith(Parameterized::class)
+class UnescapeRsWithOffsetsTest(
+    private val str: String,
+    private val decoded: String,
+    private val offsets: IntArray,
+    private val success: Boolean
+) {
+    @Test
+    fun test() {
+        val (decoded, offsets, success) = parseRustStringCharacters(str)
+
+        assertEquals(this.decoded, decoded.toString())
+        assertArrayEquals(this.offsets.copyOf(offsets.size), offsets)
+        assertEquals(this.success, success)
+    }
+
+    companion object {
+        @Parameters(name = "{index}: \"{0}\" → \"{1}\" S:{3}")
+        @JvmStatic
+        fun data(): Collection<Array<out Any>> = listOf(
+            arrayOf("", "", intArrayOf(), true),
+            arrayOf("foo", "foo", intArrayOf(0,1,2,3), true),
+            arrayOf("""a\"b""", "a\"b", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\'b""", "a\'b", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\tb""", "a\tb", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\nb""", "a\nb", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\rb""", "a\rb", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\0b""", "a\u0000b", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\\b""", "a\\b", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\x01b""", "a\u0001b", intArrayOf(0,1,5,6), true),
+            arrayOf("""a\x7Fb""", "a\u007Fb", intArrayOf(0,1,5,6), true),
+//            arrayOf("""a\x80b""", "a", intArrayOf(0,1,2), false),
```

Comment:
Let's add a comment that `"a\x80b"` is not valid rust string but now we parse it as valid string